### PR TITLE
CON-15 - DynamoDB Docker Image

### PR DESCRIPTION
**Pull request by michal-hynek**
@joshcbarnes 

This is a basic Docker image setup for DynamoDB.  I've attached the DynamoDB binary because I think it's better than having Dockerfile downloading it from internet.  I can change it if someone feels strongly against my approach.

The DB is mounted from the host computer, but I have yet to create a started DB.  I'm not sure if mounting host folder is the best approach.  An alternative is to have a separate image with a volume where the data is stored.  The only issue is that if we ever have to rebuild the data image, the volume is kinda hard to find.  I'm open to suggestions.

### CODE REVIEW

File: docker-dynamodb/docker-compose.yml [hunk 1]
```diff
@@ -0,0 +1,14 @@
+version: "3"
+
+services:
+  db:
+    image: constellation-db
+    deploy:
+      replicas: 1
+      resources:
+        cpus: 0.1
```
Comment (joshcbarnes) [inline docker-dynamodb/docker-compose.yml]: What does it mean to have 0.1 of a cpu?

File: docker-dynamodb/docker-compose.yml [hunk 2]
```diff
@@ -0,0 +1,14 @@
...
+        memory: 100M
+    volumes:
+      - ./data:/opt/dynamodb/data
```
Comment (joshcbarnes) [inline docker-dynamodb/docker-compose.yml]: Just to double check my understanding here - this is a file on the host machine that we're allowing the container to access?

So sharing dbs would amount to devs giving each other copies of this file?

**Comment by joshcbarnes** [comment]
Wow...I thought people had forgotten about this...was starting to feel a little disheartened.  What a nice monday morning surprise :D

Let's drop the docker prefix and just call this module "dynamo" within the project hierarchy - it can be a module that lives alongside the backend and frontend like you've done.  The compose file should probably be moved one level up, though.

File: docker-dynamodb/docker-compose.yml (hunk 1 above)
Comment (michal-hynek) [reply docker-dynamodb/docker-compose.yml]: It means the container shouldn't use more than 10% of the CPU's running time.

File: docker-dynamodb/docker-compose.yml (hunk 2 above)
Comment (michal-hynek) [reply docker-dynamodb/docker-compose.yml]: Yeah, the idea is to add a default DB to the repo's data folder.  So when you pull the repo, you get the default database.  After you build the image and run the container any DB updates you do locally get persisted on your computer locally.

Comment (joshcbarnes) [reply docker-dynamodb/docker-compose.yml]: I guess that means part of the build process (maybe?) should be to copy the repo's datastore file to /opt/dyanmodb/data

**Comment by michal-hynek** [comment]
Renamed `docker-dynamodb` to `dynamo` and moved `docker-compose.yml` one level up.

**Comment by joshcbarnes** [comment]
👍

**Pull request merged**